craft_parts/executor/part_handler.py

```diff
@@ -134,6 +136,8 @@ def migrate(
                     sub_path=sub_path,
                     dst_partition=dst_partition,
                 )
+                if len(sub_path) > 0:
+                    self._sub_paths_distributed.add(sub_path)
```

Comment:
If `sub_path` is not changed after creation we can move this conditional to a point right after creation. Then `sub_paths_distributed` will actually list all (non-empty) subpaths I presume?